File: LibraryBackend/Library.Models/Entities/Book.cs
using Library.Models.Entities;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Library.Services.Entities
{
    public class Book
    {
        public int Id { get; set; }
        public int AuthorId { get; set; }
        public Author Author { get; set; }
        public string Title { get; set; }
        public string Description { get; set; }
        public int PageNumber { get; set; }
        public int AvailableNumber { get; set; }
        public ICollection<Genre> Genres { get; set; } = new List<Genre>();

        public bool IsUserBook { get; set; } = false; 
        public int? UserId { get; set; } 
        public User? User { get; set; }

        public byte[]? CoverImage { get; set; }            
        public string? CoverImageContentType { get; set; }
    }
}

File: LibraryBackend/UI/elibrary_mobile/lib/models/book.dart
import 'package:elibrary_mobile/models/author.dart';
import 'package:elibrary_mobile/models/user.dart';
import 'package:json_annotation/json_annotation.dart';

import 'genre.dart';
part 'book.g.dart';

@JsonSerializable()
class Book {
  int? id;
  Author? author;
  String? title;
  String? description;
  int? pageNumber;
  int? availableNumber;
  List<Genre>? genres;

  bool? isUserBook;
  int? userId;
  User? user;

  String? coverImageBase64;
  String? coverImageContentType;

 Book(
    this.id,
    this.author,
    this.title,
    this.description,
    this.pageNumber,
    this.availableNumber,
    this.genres,
    this.isUserBook,
    this.userId,
    this.user,
    this.coverImageBase64,
    this.coverImageContentType
  );
  
  factory Book.fromJson(Map<String, dynamic> json) => _$BookFromJson(json);
  Map<String, dynamic> toJson() => _$BookToJson(this);
}